mllib/src/main/scala/org/apache/spark/ml/feature/QuantileDiscretizer.scala

```diff
@@ -114,10 +115,10 @@ final class QuantileDiscretizer @Since("1.6.0") (@Since("1.6.0") override val ui
     splits(0) = Double.NegativeInfinity
     splits(splits.length - 1) = Double.PositiveInfinity
 
-    val distinctSplits = splits.distinct
+    val distinctSplits = splits.filter(!_.isNaN).distinct
```

Comment:
Sure it could be faster, but it's also incorrect. We can't return a wrong answer here to save time. Do we disagree on the effect of including NaN when computing the quantiles?